### ISSUE TITLE
The seasonal and residual components are NaN

### ISSUE THREAD

**Issue by madhusudhankrish** [jrmontag/STLDecompose#4]
When I perform decompose(x), the returned seasonal and residual value for each observation is NaN. Could you check this? 
For the following lines of code, the output is given below.
`result=stldecompose.decompose(X)`
`print(result.seasonal) #same output for result.resid`

1998-05-06      NaN
1998-05-07      NaN
1998-05-08      NaN
1998-05-09      NaN
1998-05-10      NaN
1998-05-11      NaN
1998-05-12      NaN
1998-05-13      NaN
1998-05-14      NaN
1998-05-15      NaN
1998-05-16      NaN
1998-05-17      NaN
1998-05-18      NaN
1998-05-19      NaN
1998-05-20      NaN
1998-05-21      NaN
1998-05-22      NaN
1998-05-23      NaN
1998-05-24      NaN
1998-05-25      NaN
1998-05-26      NaN
1998-05-27      NaN
1998-05-28      NaN
1998-05-29      NaN
1998-05-30      NaN
1998-05-31      NaN

**Comment by jrmontag** [comment]
Thanks for reporting the issue, @madhusudhankrish. Can you add some additional context here? If you can include the python version you're running, and a minimum working example that illustrates the issue, I can try to help track down the issue.

**Comment by madhusudhankrish** [comment]
Python version: 3.5.3 on Anaconda. Attached a mock data file here that illustrates the issue.

    data = pd.read_csv('Data_100.csv')
    data.set_index('date', inplace=True)
    data.index = pd.to_datetime(data.index)
    current_account = data[data['319'] == account]['1998-05']
    balance_data = current_account[['balance']]
    upsampled = balance_data.resample('D')
    interpolated = upsampled.interpolate(method='linear')
    print(interpolated)
    res = stldecompose.decompose(interpolated)
    print(res.resid)
    res.plot()
    plt.show()

[Data_100.csv.tar.gz](https://github.com/jrmontag/STLDecompose/files/1448036/Data_100.csv.tar.gz)

**Comment by jrmontag** [comment]
Thanks for adding more info @madhusudhankrish. Since I think it's unwise to open random compressed tarballs online, consider including a bit more explicit data in your snippet so the whole thing is self-contained and working. 

That said, in looking at the code you've included I think you might try inspecting the data between the transformations steps - specifically the resampling and interpolating. IIRC, `.resample()` is followed by another method that indicates the operation applied in the resample e.g. `.sum()` or `.count()`.

**Comment by boegel** [comment]
@jrmontag I'm seeing the same issue (`NaN` values for `seasonal` & `resid`), with the code below and the dataset obtained from https://datamarket.com/data/set/22u3/international-airline-passengers-monthly-totals-in-thousands-jan-49-dec-60

```python
from pandas import read_csv, to_datetime
import stldecompose

series = read_csv('international-airline-passengers.csv', header=0)
series.Month = to_datetime(series.Month, errors='coerce')
series = series.set_index('Month')

result_stl = stldecompose.decompose(series)
print result_stl.seasonal
print result_stl.resid
```

The values for `result.trend` and `result.observed` look normal.

I have no problems when using `seasonal_decompose` from `statsmodels.tsa.seasonal` as a drop-in replacement for `stldecompose.decompose` in this example...

More context:

* (native) Python 2.7.10 on macOS 10.13.4
* pandas 0.22.0
* stldecompose 0.0.3
* statsmodel 0.8.0

Any idea what the problem could be? Do let me know if I can collect more information on this...

**Comment by boegel** [comment]
After digging a little deeper, it seems like I have found the issue, pull request with fix in #7
